### PR TITLE
doc: cleanup cephfs-mirroring document

### PR DESCRIPTION
**Pull request by vshankar**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by vshankar** [comment]
@mchangir I would like to include this with backport pr #39050. Please take a look.

File: doc/dev/cephfs-mirroring.rst [hunk 1]
```diff
@@ -2,9 +2,6 @@
 CephFS Mirroring
 ================
 
-.. note:: CephFS mirroring feature is currently under development. Interfaces detailed in
-          this document might change during the development cycle.
-
 CephFS will support asynchronous replication of snapshots to a remote CephFS file system via
 `cephfs-mirror` tool. Snapshots are synchronized by mirroring snapshot data followed by
 creating a snapshot with the same name (for a given directory on the remote file system) as
```
[the remark this answers is not in the document]
Comment (vshankar) [reply doc/dev/cephfs-mirroring.rst:7]: good catch

**Comment by vshankar** [comment]
@mchangir https://ceph--39242.org.readthedocs.build/en/39242/dev/cephfs-mirroring/

**Comment by vshankar** [comment]
@mchangir Pushed in few more cleanups so as to represent the current state of daemon (bulk copy, etc..).

I took out the entire section of "failed snapshots" and "Change Detection". Will readd that when we do readdir diff...

**Comment by mchangir** [comment]
> @mchangir Pushed in few more cleanups so as to represent the current state of daemon (bulk copy, etc..).
> 
> I took out the entire section of "failed snapshots" and "Change Detection". Will readd that when we do readdir diff...

oddly, I can't comment on random lines on this doc PR; only a select few around changed lines

**Comment by vshankar** [comment]
> > @mchangir Pushed in few more cleanups so as to represent the current state of daemon (bulk copy, etc..).
> > I took out the entire section of "failed snapshots" and "Change Detection". Will readd that when we do readdir diff...
> 
> oddly, I can't comment on random lines on this doc PR; only a select few around changed lines

Not sure why. Just mention the changes, I'll try to map it to the relevant section.

**Review by mchangir — APPROVED** [review]
looks good to me

**Pull request merged**